File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Java/booking/SeatFactory.java
package booking;

import utils.SeatClass;

import java.util.ArrayList;
import java.util.Map;
import java.util.List;

public class SeatFactory {
    public static List<Seat> generateSeats(Map<SeatClass, Integer> seatConfig) {
        List<Seat> seats = new ArrayList<>();
        int counter = 1;
        for (Map.Entry<SeatClass, Integer> entry : seatConfig.entrySet()) {
            for (int i = 0; i < entry.getValue(); i++) {
                seats.add(new Seat(entry.getKey().name().charAt(0) + String.valueOf(counter++), entry.getKey()));
            }
        }
        return seats;
    }
}

File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Cpp/booking/src/SeatFactory.cpp
#include "../include/SeatFactory.h"

char seatClassToChar(SeatClass seatClass) {
    switch (seatClass) {
        case SeatClass::ECONOMY: return 'E';
        case SeatClass::BUSINESS: return 'B';
        case SeatClass::FIRST: return 'F';
        default: return 'U';
    }
}

std::vector<Seat> SeatFactory::generateSeats(const std::map<SeatClass, int>& seatConfig) {
    std::vector<Seat> seats;
    int counter = 1;

    for (const auto& entry : seatConfig) {
        SeatClass seatClass = entry.first;
        int count = entry.second;
        char prefix = seatClassToChar(seatClass);

        for (int i = 0; i < count; ++i) {
            std::string seatNumber = prefix + std::to_string(counter++);
            seats.emplace_back(seatNumber, seatClass);
        }
    }

    return seats;
}